src/coreclr/jit/llvm.cpp

```diff
@@ -1654,14 +1630,190 @@ void startImportingNode(llvm::IRBuilder<>& builder)
+    BasicBlock* firstBlock = _compiler->fgFirstBB;
+    LIR::Range& firstRange = LIR::AsRange(firstBlock);
+    firstRange.InsertAtBeginning(shadowStackOffset);
+    firstRange.InsertAfter(shadowStackOffset, shadowStackArg);
+    firstRange.InsertAfter(shadowStackArg, localShadowStack);
+    firstRange.InsertAfter(localShadowStack, shadowStack);
+
+    for (BasicBlock* const block : _compiler->Blocks())
+    {
+        LIR::Range& lirRange = LIR::AsRange(block);
+        for (GenTree* node : LIR::AsRange(block))
+        {
+            genTreeOps oper = node->OperGet();
+            if (oper == GT_STORE_LCL_VAR)
+            {
+                GenTreeLclVarCommon* lclVar = node->AsLclVarCommon();
+                LclVarDsc* varDsc = _compiler->lvaGetDesc(lclVar->GetLclNum());
+                if (!varDsc->lvIsParam && !canStoreLocalOnLlvmStack(varDsc))
+                {
+                    GenTreeIntCon* offset;
+                    GenTreeLclVar* shadowStackVar;
+                    GenTree* addNode = createAddNodeForShadowStackLocal(varDsc, shadowStackLclNum, offset, shadowStackVar);
+
+                    genTreeOps oper = lclVar->TypeGet() == var_types::TYP_STRUCT ? GT_STORE_OBJ : GT_STOREIND;
+
+                    node->ChangeOper(oper);
+                    node->gtFlags |= GTF_IND_TGT_NOT_HEAP;
+                    GenTreeOp* opNode = node->AsOp();
+                    opNode->gtOp2 = opNode->gtGetOp1();
+                    opNode->gtOp1 = addNode;
+
+                    insertAddWithOperands(lirRange, node, addNode, offset, shadowStackVar);
+                }
+            }
+            else if (oper == GT_LCL_VAR)
+            {
+                GenTreeLclVarCommon* lclVar = node->AsLclVarCommon();
+                LclVarDsc* varDsc = _compiler->lvaGetDesc(lclVar->GetLclNum());
+                if (!varDsc->lvIsParam && !canStoreLocalOnLlvmStack(varDsc))
+                {
+                    GenTreeIntCon* offset;
+                    GenTreeLclVar* shadowStackVar;
+                    GenTree* addNode = createAddNodeForShadowStackLocal(varDsc, shadowStackLclNum, offset, shadowStackVar);
+
+                    genTreeOps oper = lclVar->TypeGet() == var_types::TYP_STRUCT ? GT_OBJ : GT_IND;
+
+                    node->ChangeOper(oper);
+                    node->AsOp()->gtOp1 = addNode;
+
+                    insertAddWithOperands(lirRange, node, addNode, offset, shadowStackVar);
+                }
+            }
+        }
+    }
+
+    if (_compiler->verboseTrees)
+    {
+        _compiler->fgDispBasicBlocks(true);
+    }
+}
+
+//------------------------------------------------------------------------
+// Convert GT_STORE_LCL_VAR and GT_LCL_VAR to use the shadow stack when the local needs to be GC tracked
+//
+void Llvm::PlaceAndConvertShadowStackLocals()
+{
+    _shadowStackLocalsSize = 0;
+    if (_compiler->lvaCount == 0) return;
```

Comment:
```suggestion
    if (_compiler->lvaCount == 0)
    {
        return;
    }
```
A rare condition, but presumably can happen.